posts/tips-and-tricks-for-manually-scaling-a-global-dynamodb-table-from-an-aws-lambda.md

```diff
@@ -0,0 +1,128 @@
+- Initialize
+```java
+private DynamoDbClient getClient() {
+    ApacheHttpClient.Builder httpClientBuilder = ApacheHttpClient.builder()
+            .connectionTimeout(CONNECTION_TIMEOUT)
+            .socketTimeout(SOCKET_TIMEOUT);
+    RetryPolicy.Builder retryBuilder = RetryPolicy.builder()
+            .numRetries(RETRY_LIMIT);
+    ClientOverrideConfiguration.Builder clientOverrideBuilder =
+            ClientOverrideConfiguration.builder()
+                    .apiCallAttemptTimeout(API_CALL_ATTEMPT_TIMEOUT)
+                    .apiCallTimeout(API_CALL_TIMEOUT)
+                    .retryPolicy(retryBuilder.build());
+
+    return DynamoDbClient.builder()
+            .httpClientBuilder(httpClientBuilder)
+            .overrideConfiguration(clientOverrideBuilder.build())
+            .build();
+}
+```
+- Call the SDK
+```java
+AutoScalingTargetTrackingScalingPolicyConfigurationUpdate policyConfigurationUpdate =
+        AutoScalingTargetTrackingScalingPolicyConfigurationUpdate.builder()
+                .targetValue(SCALING_POLICY_TARGET_VALUE)
+                .build();
+AutoScalingPolicyUpdate policyUpdate = AutoScalingPolicyUpdate.builder()
+        .targetTrackingScalingPolicyConfiguration(policyConfigurationUpdate)
+        .build();
+AutoScalingSettingsUpdate.Builder updateBuilder = AutoScalingSettingsUpdate.builder()
+        .maximumUnits(MAX_AUTOSCALE_CAPACITY)
+        .scalingPolicyUpdate(policyUpdate);
+
+AutoScalingSettingsUpdate rcuUpdate = updateBuilder.minimumUnits(HIGH_READ_CAPACITY_UNITS).build();
+AutoScalingSettingsUpdate wcuUpdate = updateBuilder.minimumUnits(HIGH_WRITE_CAPACITY_UNITS).build();
+
+ReplicaAutoScalingUpdate eastReplicaUpdate = ReplicaAutoScalingUpdate.builder()
+        .regionName("us-east-1")
+        .replicaProvisionedReadCapacityAutoScalingUpdate(rcuUpdate)
+        .build();
+ReplicaAutoScalingUpdate westReplicaUpdate = ReplicaAutoScalingUpdate.builder()
+        .regionName("us-west-2")
+        .replicaProvisionedReadCapacityAutoScalingUpdate(rcuUpdate)
+        .build();
+UpdateTableReplicaAutoScalingRequest autoScalingRequest = UpdateTableReplicaAutoScalingRequest.builder()
+        .tableName(TABLE_NAME)
+        .replicaUpdates(eastReplicaUpdate, westReplicaUpdate)
+        .provisionedWriteCapacityAutoScalingUpdate(wcuUpdate)
+        .build();
+
+try {
+    getClient().updateTableReplicaAutoScaling(autoScalingRequest);
+} catch (AmazonServiceException e) {
+    System.err.println(e.getErrorMessage());
+    System.exit(1);
+}
+```
+
+# Issues
+The code above worked great until the Lambda was deployed in _us-west-2_ for redundancy. The SDK calls from _us-west-2_ were timing out.
+- For testing purposes, no SDK v2 calls worked but v1 calls did
+- The solution is to initialize the SDK v1 and then make the SDK v2 calls
+- Add a call to `getTable(TABLE_NAME)` from the SDK v1 before the call to `updateTableReplicaAutoScaling`
+- **Note:** SDK v1 needs the proxy specified but SDK v2 does not
```

Comment:
If it's necessary to use both SDK v1 and v2, it may be worthwhile to briefly discuss the differences between the two.  Under what scenario would you use v1 vs. v2 or what makes v2 more powerful than v1 etc.